File: src/Deviser.Core/Deviser.Core.Data/Entities/Property.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Deviser.Core.Data.Entities
{
    public class Property
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Label { get; set; }
        //Value property is not in DB, It is here to maintain JSON strucuture
        public string Value { get; set; }
        public string DefaultValue { get; set; }
        public string Description { get; set; }
        public Guid? OptionListId { get; set; }
        public ICollection<LayoutTypeProperty> LayoutTypeProperties { get; set; }
        public ICollection<ContentTypeProperty> ContentTypeProperties { get; set; }
        public ICollection<ModuleViewProperty> ModuleViewProperties { get; set; }
        public OptionList OptionList { get; set; }
        public bool IsActive { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? LastModifiedDate { get; set; }

    }
}

File: src/Deviser.Web/ClientApp/src/app/common/domain-types/property.ts
import { OptionList } from './option-list';
import { PropertyOption } from './property-option';

export interface Property
    {
        id: string;
        name: string;
        label: string;
        //Value property is not in DB, It is here to maintain JSON strucuture
        nalue: string;
        defaultValue: string;
        description: string;
        optionListId: string | null;        
        optionList: OptionList;
        defaultValuePropertyOption: PropertyOption;
        isActive: boolean;
        createdDate: Date | string | null;
        lastModifiedDate: Date | string | null;
    }